### PR TITLE
EV: Fixed allocation

### PR DESCRIPTION
**Pull request by REASY**
Run on EV-fleet.conf with 5 iteration:
Only this branch: https://drive.google.com/open?id=1Mm7OCJbdCCQgl6KZ978_aG2jQfFeYWe1
With zonal parking optimization: https://drive.google.com/open?id=1wjdqxzLon_3ADtMz3OYdqOkx5Xtk0Tz4

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/780)

### CODE REVIEW

File: src/main/scala/beam/agentsim/agents/ridehail/allocation/EVFleetAllocationManager.scala [hunk 1]
```diff
@@ -62,16 +60,18 @@ class EVFleetAllocationManager(val rideHailManager: RideHailManager)
         // from future consideration if it fails this check
         val routingResponses = routeResponsesByDriver.get(agentLocation.vehicleId)
 
+        val routedDisanceInM = routingResponses
+          .getOrElse(List[RoutingResponse]())
```
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/agents/ridehail/allocation/EVFleetAllocationManager.scala]: Why not just make this a fold also?

**Comment by REASY** [comment]
[QuadTree.getDisk](https://github.com/matsim-org/matsim/blob/0.10.x/matsim/src/main/java/org/matsim/core/utils/collections/QuadTree.java#L140) uses `ArrayList`. We can try to change it to LinkedList because basically it's used as append only collection. 
image

File: src/main/scala/beam/agentsim/agents/ridehail/allocation/EVFleetAllocationManager.scala (hunk 1 above)
Comment (colinsheppard) [reply src/main/scala/beam/agentsim/agents/ridehail/allocation/EVFleetAllocationManager.scala]: cause I'm not a scala ninja :)

Comment (JustinPihony) [reply src/main/scala/beam/agentsim/agents/ridehail/allocation/EVFleetAllocationManager.scala]: Fair point! I should word my comments better :)

**Pull request merged**